packages/venia-concept/src/index.js

```diff
@@ -51,21 +51,24 @@ ReactDOM.render(
     document.getElementById('root')
 );
 
-if (
-    process.env.NODE_ENV === 'production' ||
-    process.env.DEV_SERVER_SERVICE_WORKER_ENABLED
-) {
-    window.addEventListener('load', () =>
-        navigator.serviceWorker
-            .register('/sw.js')
-            .then(registration => {
-                console.log('Service worker registered: ', registration);
-            })
-            .catch(error => {
-                console.log('Service worker registration failed: ', error);
-            })
-    );
-}
+// This is stored as a global for testing purposes.
+window.__registerServiceWorker = () => {
```

Comment:
Looks like this PR changes things, that have been moved to different files as part of #1905

Reply:
Not a fan of custom globals though `__` prefix is OK I guess.

Is there not another way around this?

Reply:
It's ugly, but it's incredibly common to have a custom global in an app like this one that takes over the whole document. And it does have the advantage that something else can remotely unsubscribe the handler, because there's a real reference to the handler: `window.removeEventListener('load', window.__registerServiceWorker)` is something that a future Webpack hot module reloading strategy could use.

Reply:
@revanth0212 Once one of these is merged, we'll rebase the other one!